addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComUndecodedRFMessage.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rfxcom.internal.messages;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.eclipse.smarthome.core.library.items.StringItem;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.openhab.binding.rfxcom.RFXComValueSelector;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComException;
+
+/**
+ * RFXCOM data class for undecoded messages.
+ *
+ * @author James Hewitt-Thomas
+ */
+public class RFXComUndecodedRFMessage extends RFXComBaseMessage {
+
+    public enum SubType {
+        AC(0x00),
+        ARC(0x01),
+        ATI(0x02),
+        HIDEKI_UPM(0x03),
+        LACROSSE_VIKING(0x04),
+        AD(0x05),
+        MERTIK(0x06),
+        OREGON1(0x07),
+        OREGON2(0x08),
+        OREGON3(0x09),
+        PROGUARD(0x0a),
+        VISONIC(0x0b),
+        NEC(0x0c),
+        FS20(0x0d),
+        RESERVED(0x0e),
+        BLINDS(0x0f),
+        RUBICSON(0x10),
+        AE(0x11),
+        FINE_OFFSET(0x12),
+        RGB(0x13),
+        RTS(0x14),
+        SELECT_PLUS(0x15),
+        HOME_CONFORT(0x16),
+
+        UNKNOWN(0xFF);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        SubType(byte subType) {
+            this.subType = subType;
+        }
+
+        public byte toByte() {
+            return (byte) subType;
+        }
+
+        public static SubType fromByte(int input) {
```

Comment:
Haha, I nabbed that from your pending pull request, because I'm hoping it goes in eventually too.